### ISSUE TITLE
[Windows] Library not found : hunspell

### ISSUE THREAD

**Issue by CPlusPlus17** [valentjn/vscode-ltex#40]
**Describe the bug**
Spell checking is not working because of missing hunspell lib.

**Steps to reproduce**
Open any file for spell checking.

**LTeX configuration**
Default, with de-CH set.

**Log**

> May 22, 2020 4:43:49 PM org.bsplines.languagetool_languageserver.LanguageToolLanguageServer initialize
> INFO: Setting locale to en
> SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
> SLF4J: Defaulting to no-operation (NOP) logger implementation
> SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
> Warning: At the moment, your platform (Windows) is not supported by the official XGBoost maven package; ML-based suggestion reordering is disabled.
> WARN: ngram index dir \de not found for German (Swiss)
> May 22, 2020 4:43:52 PM org.bsplines.languagetool_languageserver.LanguageToolLanguageServer lambda$validateDocument$3
> INFO: Checking the following text in language de-CH via LanguageTool: " Einleitung. Welche betriebswirtschaftlichen und allgemeinen Vorteile ergeben sich durch Digitalisie"... (truncated to 100 characters)
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell
> May 22, 2020 4:43:54 PM org.bridj.BridJ log
> INFO: Library not found : hunspell

**Version information**

- Operating system: Windows 10 latest
- VS Code: latest
- LTeX: latest
- LTeX language extensions: german
- Java: latest 8

**Comment by valentjn** [comment]
https://forum.languagetool.org/t/announcement-languagetool-4-8/5213/7 says this might occur on 32-bit machines. Are you on 32-bit or 64-bit (check in Windows Settings -> System -> About or Info)?

**Comment by CPlusPlus17** [comment]
Hello

Windows 10 64-bit. Confirmed on two installations. Did some regression testing, bug started with 4.9.0. The 4.7.x branch works without problems.

Greets
Manuel

**Comment by valentjn** [comment]
LT upgraded the embedded hunspell in 4.8. LT<sub>E</sub>X skipped LT 4.8, so the first LT<sub>E</sub>X version with LT >= 4.8 is 4.9.0. Therefore, I'd say this is an LT issue.

Unfortunately, I can't reproduce this with LT<sub>E</sub>X 4.9.3, German language support extension 4.9.0, and `settings.json` = ``{"ltex.language": "de-CH"}``:

```log
Mai 22, 2020 6:33:58 NACHM. org.bsplines.languagetool_languageserver.LanguageToolLanguageServer initialize
INFORMATION: Setting locale to en
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Mai 22, 2020 6:33:59 NACHM. org.bsplines.languagetool_languageserver.LanguageToolLanguageServer reinitialize
SCHWERWIEGEND: en-US is not a recognized language. Leaving LanguageTool uninitialized, checking disabled.
Warning: At the moment, your platform (Windows) is not supported by the official XGBoost maven package; ML-based suggestion reordering is disabled.
Mai 22, 2020 6:34:04 NACHM. org.bsplines.languagetool_languageserver.LanguageToolLanguageServer lambda$validateDocument$3
INFORMATION: Checking the following text in language de-CH via LanguageTool: "Einleitung. Welche betriebswirtschaftlichen und allgemeinen Vorteile ergeben sich durch.\n\n"
Mai 22, 2020 6:34:08 NACHM. org.bsplines.languagetool_languageserver.LanguageToolLanguageServer lambda$validateDocument$3
INFORMATION: Obtained 0 rule matches
```

I also get spelling suggestions, e.g., if I misspell `durch` as `druch`.

Can you reproduce the bug with LT standalone? Download https://languagetool.org/download/LanguageTool-4.9.zip, write some short text in `test.txt`, and run `java -jar languagetool-commandline.jar --language de-CH test.txt`.

**Comment by CPlusPlus17** [comment]
Seems you are right:

```
PS C:\Users\manue\Downloads\LanguageTool-4.9.1\LanguageTool-4.9.1> java -jar languagetool-commandline.jar --language de-CH test.txt
Warning: At the moment, your platform (Windows) is not supported by the official XGBoost maven package; ML-based suggestion reordering is disabled.
Expected text language: German (Swiss)
Working on test.txt...
May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

May 22, 2020 8:27:37 PM org.bridj.BridJ log
INFO: Library not found : hunspell
# BridJ: LoadLibrary error when loading hunspell : The specified module could not be found.

Exception in thread "main" java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.UnsatisfiedLinkError
        at org.languagetool.MultiThreadedJLanguageTool.performCheck(MultiThreadedJLanguageTool.java:175)
        at org.languagetool.JLanguageTool.check(JLanguageTool.java:820)
        at org.languagetool.JLanguageTool.check(JLanguageTool.java:784)
        at org.languagetool.JLanguageTool.check(JLanguageTool.java:775)
        at org.languagetool.JLanguageTool.check(JLanguageTool.java:758)
        at org.languagetool.JLanguageTool.check(JLanguageTool.java:717)
        at org.languagetool.JLanguageTool.check(JLanguageTool.java:701)
        at org.languagetool.commandline.CommandLineTools.checkText(CommandLineTools.java:106)
        at org.languagetool.commandline.CommandLineTools.checkText(CommandLineTools.java:82)
        at org.languagetool.commandline.Main.runOnFile(Main.java:195)
        at org.languagetool.commandline.Main.main(Main.java:467)
Caused by: java.util.concurrent.ExecutionException: java.lang.UnsatisfiedLinkError
        at java.util.concurrent.ForkJoinTask.get(Unknown Source)
        at org.languagetool.MultiThreadedJLanguageTool.performCheck(MultiThreadedJLanguageTool.java:172)
        ... 10 more
Caused by: java.lang.UnsatisfiedLinkError
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at java.util.concurrent.ForkJoinTask.getThrowableException(Unknown Source)
        ... 12 more
Caused by: java.lang.UnsatisfiedLinkError: dumonts.hunspell.bindings.HunspellLibrary.Hunspell_create(JJ)J
        at dumonts.hunspell.bindings.HunspellLibrary.Hunspell_create(Native Method)
        at dumonts.hunspell.bindings.HunspellLibrary.Hunspell_create(HunspellLibrary.java:25)
        at org.languagetool.rules.spelling.hunspell.Hunspell.<init>(Hunspell.java:46)
        at org.languagetool.rules.spelling.hunspell.Hunspell.getInstance(Hunspell.java:59)
        at org.languagetool.rules.spelling.hunspell.HunspellRule.init(HunspellRule.java:379)
        at org.languagetool.rules.de.GermanSpellerRule.init(GermanSpellerRule.java:922)
        at org.languagetool.rules.de.SwissGermanSpellerRule.init(SwissGermanSpellerRule.java:56)
        at org.languagetool.rules.spelling.hunspell.HunspellRule.match(HunspellRule.java:143)
        at org.languagetool.JLanguageTool.checkAnalyzedSentence(JLanguageTool.java:944)
        at org.languagetool.JLanguageTool$TextCheckCallable.getOtherRuleMatches(JLanguageTool.java:1414)
        at org.languagetool.JLanguageTool$TextCheckCallable.call(JLanguageTool.java:1345)
        at org.languagetool.JLanguageTool$TextCheckCallable.call(JLanguageTool.java:1306)
        at java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(Unknown Source)
        at java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(Unknown Source)
        at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
        at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
```

**Comment by valentjn** [comment]
Then it's an LT issue. I'd suggest you open an issue over at https://github.com/languagetool-org/languagetool. If it's indeed a bug in LT that gets a fix, the next LT<sub>E</sub>X release will contain the fixed LT version. Also, LT<sub>E</sub>X will soon have the possibility to use an external LT server, so you will be able to use your own LT version, e.g., 4.7 as that still works for you.

PS: I took the liberty of formatting your log output a bit better. Hope you don't mind.

**Comment by valentjn** [comment]
As mentioned above, the newly released version 5.0.0 of LT<sub>E</sub>X can connect to an LT HTTP server, so you can downgrade LT without having to downgrade LT<sub>E</sub>X.
